### PR TITLE
fixed issue #2154 for achievements activity

### LINKED ISSUE #2065: Achievements activity crashes on API 19
**Summary:** 
Achievements activity crashes for API <=19 , can be related with #1930

**Steps to reproduce:** 
Click your username
**System logs:**

```
  java.lang.RuntimeException: Unable to start activity ComponentInfo{fr.free.nrw.commons.beta/fr.free.nrw.commons.achievements.AchievementsActivity}: android.content.res.Resources$NotFoundException: File res/drawable/badge.xml from drawable resource ID #0x7f08005b
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2338)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2390)
        at android.app.ActivityThread.access$800(ActivityThread.java:151)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1321)
        at android.os.Handler.dispatchMessage(Handler.java:110)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:5292)
        at java.lang.reflect.Method.invokeNative(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:515)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:824)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:640)
        at dalvik.system.NativeStart.main(Native Method)
     Caused by: android.content.res.Resources$NotFoundException: File res/drawable/badge.xml from drawable resource ID #0x7f08005b
        at android.content.res.Resources.loadDrawable(Resources.java:2152)
        at android.content.res.Resources.getDrawable(Resources.java:710)
        at android.support.v7.widget.ResourcesWrapper.getDrawable(ResourcesWrapper.java:129)
        at android.support.v7.widget.TintResources.getDrawable(TintResources.java:45)
        at android.support.v4.content.ContextCompat.getDrawable(ContextCompat.java:360)
        at android.support.v7.widget.AppCompatDrawableManager.getDrawable(AppCompatDrawableManager.java:198)
        at android.support.v7.widget.AppCompatDrawableManager.getDrawable(AppCompatDrawableManager.java:186)
        at android.support.v7.content.res.AppCompatResources.getDrawable(AppCompatResources.java:100)
        at android.support.v7.widget.AppCompatImageHelper.setImageResource(AppCompatImageHelper.java:85)
        at android.support.v7.widget.AppCompatImageView.setImageResource(AppCompatImageView.java:93)
        at fr.free.nrw.commons.achievements.AchievementsActivity.onCreate(AchievementsActivity.java:122)
        at android.app.Activity.performCreate(Activity.java:5264)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1088)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2302)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2390) 
        at android.app.ActivityThread.access$800(ActivityThread.java:151) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1321) 
        at android.os.Handler.dispatchMessage(Handler.java:110) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:5292) 
        at java.lang.reflect.Method.invokeNative(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:515) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:824) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:640) 
        at dalvik.system.NativeStart.main(Native Method) 
     Caused by: org.xmlpull.v1.XmlPullParserException: Binary XML file line #1: invalid drawable tag vector
        at android.graphics.drawable.Drawable.createFromXmlInner(Drawable.java:933)
        at android.graphics.drawable.Drawable.createFromXml(Drawable.java:877)
        at android.content.res.Resources.loadDrawable(Resources.java:2148)
        at android.content.res.Resources.getDrawable(Resources.java:710) 
        at android.support.v7.widget.ResourcesWrapper.getDrawable(ResourcesWrapper.java:129) 
        at android.support.v7.widget.TintResources.getDrawable(TintResources.java:45) 
        at android.support.v4.content.ContextCompat.getDrawable(ContextCompat.java:360) 
        at android.support.v7.widget.AppCompatDrawableManager.getDrawable(AppCompatDrawableManager.java:198) 
        at android.support.v7.widget.AppCompatDrawableManager.getDrawable(AppCompatDrawableManager.java:186) 
        at android.support.v7.content.res.AppCompatResources.getDrawable(AppCompatResources.java:100) 
        at android.support.v7.widget.AppCompatImageHelper.setImageResource(AppCompatImageHelper.java:85) 
        at android.support.v7.widget.AppCompatImageView.setImageResource(AppCompatImageView.java:93) 
        at fr.free.nrw.commons.achievements.AchievementsActivity.onCreate(AchievementsActivity.java:122) 
        at android.app.Activity.performCreate(Activity.java:5264) 
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1088) 
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2302) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2390) 
        at android.app.ActivityThread.access$800(ActivityThread.java:151) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1321) 
        at android.os.Handler.dispatchMessage(Handler.java:110) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:5292) 
        at java.lang.reflect.Method.invokeNative(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:515) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:824) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:640) 
        at dalvik.system.NativeStart.main(Native Method) 
12-03 21:16:51.770 674-997/? E/Sensors: new acc setDelay handle(0),ns(66667000) err! go to hwmsen
```

**Device and Android version:** 

API 19 or lover

### LINKED ISSUE #2154: Achievements Activity has no UI for new user
**Issue in Achievements Activity**
When user has made no contributions to share information about places. Then achievements activity should prepare different UI for such people mentioning there level and also clearly state that the user has made no contributions instead it is noticed that Achievemnts activity has no prepared UI for such a case also it shows a Toast message saying that an error has occured which is inappropriate.

**Steps to reproduce:** 

How can we reproduce the issue? 
Only those user's who have't made any contributions to share information of places can see this issue.
If you haven't made any contributions then go to Achievements activity via Navigation Drawer. And you'll find an empty activity which should't be noticed because achievement activity has it's own prepared UI. 

**What did you expect the app to do, and what did you see instead?**

App should  clearly state that the user has made no contributions instead it is noticed that Achievemnts activity has no prepared UI for such a case also it shows a Toast message saying that an error has occurred which is inappropriate.

Device- vivo 1713, Android version-7.0 Nougat API Level-24

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by vanshikaarora**
Achievements activity faced frequent crashes due to different json response delivered from backened when hasn't made any contributions. So in this case json response recieved was unexpected hence generated crashes. 

Fixes #2154 Achievements activity has no UI for new user who contributed zero contributions.

To solve the error i made changes to
1. ApacheHttpClientMediaWikiApi.java to deliver a specific response in case of crash due to json.
2. AchievementsActivity.java to interpret the specific result delivered by getAchievements function of ApacheHttpClientMediaWikiApi and to modify the code and ui accordingly.

Need help? See https://support.google.com/android/answer/9075928

---

_Note: Please ensure that you have read CONTRIBUTING.md if this is your first pull request._

### CODE REVIEW

**Comment by maskaravivek** [comment]
@vanshikaarora Can you paste the JSON response that you received for your account?

**Comment by vanshikaarora** [comment]
> @vanshikaarora Can you paste the JSON response that you received for your account?

@maskaravivek  the json response was Content-type: application/json which is inappropriate because the type of response expected is an object of type FeedbackResponse.

**Comment by maskaravivek** [comment]
@vanshikaarora  `Content-type: application/json` is the type of data returned by the API. The actual JSON response would be something that could be parsed to an object of type `FeedbackResponse`. 

Can you share your MediaWiki username so that we can check the response for your ID or else can you share logcat for the same.

Am curious because the issue that @neslihanturan posted refers to some error in parsing the xml badge icon. 

```
Caused by: org.xmlpull.v1.XmlPullParserException: Binary XML file line #1: invalid drawable tag vector
```

**Comment by vanshikaarora** [comment]
> @vanshikaarora `Content-type: application/json` is the type of data returned by the API. The actual JSON response would be something that could be parsed to an object of type `FeedbackResponse`.
> 
> Can you share your MediaWiki username so that we can check the response for your ID or else can you share logcat for the same.
> 
> Am curious because the issue that @neslihanturan posted refers to some error in parsing the xml badge icon.
> 
> ```
> Caused by: org.xmlpull.v1.XmlPullParserException: Binary XML file line #1: invalid drawable tag vector
> ```

@maskaravivek sir the error that i had recieved was "Expected BEGIN_OBJECT but was a string at line 1 and column 1" so that was probably an error occuring at getAchievements() in ApacheHttpClientMediaWikiApi.java so this error was definitely due to unexpected type response response recieved from json. Hence I modified the code to handle this end case.

**Comment by maskaravivek** [comment]
If you received a different error then this PR should be not linked with the existing issue. Feel free to create a new issue for the error you received and submit this PR for that issue instead. API level 19 might need an additional fix to make it work. :)

**Comment by vanshikaarora** [comment]
> If you received a different error then this PR should be not linked with the existing issue. Feel free to create a new issue for the error you received and submit this PR for that issue instead. API level 19 might need an additional fix to make it work. :)

@maskaravivek  didn't you got this same error in achievements activity when yo have no contributions?

**Comment by maskaravivek** [comment]
This feature was added recently and i already contributions till then. If you ahare your userName i can check the response for your user

**Comment by vanshikaarora** [comment]
> This feature was added recently and i already had contributions then. If you ahare your userName i can check the response for your user

my username is Vanshika

**Comment by maskaravivek** [comment]
@vanshikaarora Thanks. Yes, it seems the JSON is invalid. Can you create a new issue for this fix? 
https://github.com/commons-app/apps-android-commons/issues/new

The original issue #2065 still needs to be fixed.

**Comment by vanshikaarora** [comment]
> @vanshikaarora Thanks. Yes, it seems the JSON is invalid. Can you create a new issue for this fix?
> https://github.com/commons-app/apps-android-commons/issues/new
> 
> The original issue #2065 still needs to be fixed.

Sure @maskaravivek I'll create a new issue.
Well also could you please let me know whether the changes that i had made in code are appropriate for the issue or not. Or should i need to try different methods too.

Regards Vanshika Arora

**Comment by vanshikaarora** [comment]
Thank you sir i have  created a new issue here
https://github.com/commons-app/apps-android-commons/issues/2154
Now you my PR can be merged or you should i send another PR.

Since I am new to open source. It would be a  great pleasure to hear from
you. For valuable suggestions and advice.


Thanks and Regards
Vanshika Arora
Sophomre at IIT Dhanbad.

**Comment by vanshikaarora** [comment]
Sir I am new to open source could you please Let me know why isn't my PR
accepted for the work.
Eager to hear from you

Thanks and Regards
Vanshika Arora

**Comment by neslihanturan** [comment]
@vanshikaarora which PR and what you mean by "is not accepted"?

**Comment by vanshikaarora** [comment]
The PR posted here https://github.com/commons-app/apps-android-commons/pull/2087

**Comment by neslihanturan** [comment]
When you read PR description you wrote, you will see the description and function do not match. This PR supposed to solve #2065 but it attempt to solve #2154 . So my suggestion is closing this PR and sending new one for #2154 or editing title and description of this PR accordingly.

**Comment by vanshikaarora** [comment]
Hey @neslihanturan I have made changes to the PR title to point at the correct issue. Now can you please look into this and accept my Pull request.
Thanks for the help

**Comment by neslihanturan** [comment]
Description still points to wrong issue, I am fixing though.

File: app/build.gradle [hunk 1]
```diff
@@ -201,5 +201,5 @@ android {
         targetCompatibility JavaVersion.VERSION_1_8
     }
 
-    buildToolsVersion buildToolsVersion
+    buildToolsVersion '28.0.3'
 }
```
Comment (neslihanturan) [inline app/build.gradle]: This issue shouldn't etid build.gradle file, since this is irrelevant. Please revert this change.

**Comment by vanshikaarora** [comment]
So sorry for that
Well, I have made the necessary changes. Please can you consider it.

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java [hunk 2]
```diff
@@ -186,23 +197,42 @@ void shareScreen(Bitmap bitmap) {
...
+                                response -> {
+                                    if (response != null) {
+                                        setUploadCount(Achievements.from(response));
+                                    } else {
+                                        Timber.d("success");
+                                        layoutImageReverts.setVisibility(View.INVISIBLE);
+                                        imageView.setVisibility(View.INVISIBLE);
+                                        levelNumber.setText("You haven't qualified any level yet");
+                                        //onError();
+                                    }
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: Please clear comment outed lines.

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java [hunk 3]
```diff
@@ -186,23 +197,42 @@ void shareScreen(Bitmap bitmap) {
...
-                    .subscribeOn(Schedulers.io())
-                    .observeOn(AndroidSchedulers.mainThread())
-                    .subscribe(
-                            response -> {
-                                if (response != null) {
-                                    setUploadCount(Achievements.from(response));
-                                } else {
+            //Timber.d(String.valueOf(mediaWikiApi.getAchievements(sessionManager.getCurrentAccount().name)));
+                /*layoutImageReverts.setVisibility(View.VISIBLE);
+                imageView.setVisibility(View.VISIBLE);*/
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: Please clear also there comments, since they are not descriptive

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java [hunk 4]
```diff
@@ -186,23 +197,42 @@ void shareScreen(Bitmap bitmap) {
...
-                            t -> {
-                                Timber.e(t, "Fetching achievements statistics failed");
-                                onError();
-                            }
-                    ));
+                        ));
+            }
+            catch (Exception e){
+                Timber.d(e+"success");
+                    /*layoutImageReverts.setVisibility(View.INVISIBLE);
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: Please clear this comments, they are not descriptive.

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java [hunk 5]
```diff
@@ -247,10 +277,38 @@ private void setAchievementsUploadCount(Achievements achievements, int uploadCou
...
+        imagesUploadedProgressbar.setVisibility(View.INVISIBLE);
+        imageRevertsProgressbar.setVisibility(View.INVISIBLE);
+        imagesUsedByWikiProgessbar.setVisibility(View.INVISIBLE);
+        //levelNumber.setText("LEVEL 0");
+        imageView.setVisibility(View.INVISIBLE);
+        imageByWikiText.setText("No images used");
+        imageRevertedText.setText("No image reverted");
+        imageUploadedText.setText("No images Uploaded");
+        /*BitmapDrawable bitmapImage = BitmapUtils.writeOnDrawable(bitmap, Integer.toString(levelInfo.getLevelNumber()),this);
+        imageView.setImageDrawable(bitmapImage);*/
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: Please remove these comments.

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java [hunk 6]
```diff
@@ -315,6 +373,7 @@ private void hideProgressBar(Achievements achievements) {
             layoutStatistics.setVisibility(View.VISIBLE);
             imageView.setVisibility(View.VISIBLE);
             levelNumber.setVisibility(View.VISIBLE);
+            //levelNumber.setText("You haven't qualified any level yet");
         }
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: Also this comment.

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java [hunk 7]
```diff
@@ -388,10 +447,10 @@ private void launchAlert(String title, String message){
     private boolean checkAccount(){
         Account currentAccount = sessionManager.getCurrentAccount();
         if (currentAccount == null) {
-        Timber.d("Current account is null");
-        ViewUtil.showLongToast(this, getResources().getString(R.string.user_not_logged_in));
-        sessionManager.forceLogin(this);
-        return false;
+            Timber.d("Current account is null");
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: This changes are valid but irrelevant with this PR, please revert them.

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java [hunk 8]
```diff
@@ -247,10 +277,38 @@ private void setAchievementsUploadCount(Achievements achievements, int uploadCou
...
+        dialog.show();
+        imagesUploadedProgressbar.setVisibility(View.INVISIBLE);
+        imageRevertsProgressbar.setVisibility(View.INVISIBLE);
+        imagesUsedByWikiProgessbar.setVisibility(View.INVISIBLE);
+        //levelNumber.setText("LEVEL 0");
+        imageView.setVisibility(View.INVISIBLE);
+        imageByWikiText.setText("No images used");
+        imageRevertedText.setText("No image reverted");
+        imageUploadedText.setText("No images Uploaded");
+        /*BitmapDrawable bitmapImage = BitmapUtils.writeOnDrawable(bitmap, Integer.toString(levelInfo.getLevelNumber()),this);
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: All strings should be in strings.xml file

File: app/src/main/java/fr/free/nrw/commons/mwapi/ApacheHttpClientMediaWikiApi.java [hunk 9]
```diff
@@ -988,7 +989,15 @@ public boolean isUserBlockedFromCommons() {
...
                 }
-                return gson.fromJson(json, FeedbackResponse.class);
+                Timber.d(json);
+                try {
+                    return gson.fromJson(String.valueOf(response.body()), FeedbackResponse.class);
+                }
+                catch (Exception e){
+                    return new FeedbackResponse("",0,0,0,new FeaturedImages(0,0),0,"",0);
+                }
+
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/mwapi/ApacheHttpClientMediaWikiApi.java]: Can you please explain why did you add this part?

File: gradle/wrapper/gradle-wrapper.properties [hunk 10]
```diff
@@ -1,6 +1,6 @@
-#Sat Nov 03 11:02:51 CET 2018
+#Sat Dec 08 21:00:44 IST 2018
```
Comment (neslihanturan) [inline gradle/wrapper/gradle-wrapper.properties]: Please do not edit this file.

**Comment by vanshikaarora** [comment]
Are you talking about gradle-wrapper.properties?

**Comment by neslihanturan** [comment]
Hi @vanshikaarora , I have added some reviews, please make requested changes. Somethings you need to check while sending PR:
- Make sure you just changed required lines to fix your issue, no more.
- Make sure you cleaned all irrelevant imports and comment outs.
- Make sure you didn't touch gradle files unless required, generally Android Studio offers, you should say no.
- Make sure you checked "FilesChanged" tab on github. Then control all lines, these are the changes you send to code base, and you need to make sure all of them are valid. No extra fixes under 1 PR. 

Waiting updates from you :)

File: gradle/wrapper/gradle-wrapper.properties (hunk 10 above)
Comment (neslihanturan) [reply gradle/wrapper/gradle-wrapper.properties]: You can reply inline from github interface, yes I am talking about the file I wrote my inline comment.

**Comment by vanshikaarora** [comment]
Sure I'll do that soon

**Comment by vanshikaarora** [comment]
Hey @neslihanturan I have made the necessary changes for removing unwanted comments and restoring contents of gradle files.

**Comment by vanshikaarora** [comment]
Also in respone to your question for changes made to ApacheHttpClientMediaWikiApi.java: 
1.The app crashes whenever user has made no contribution and that was because the json response returned in such a case is just "applications/json" and cannot be converted to type FeedbackResponse.
So in such case the gson to json converter returns an error. Hence i wrapped the thing around try catch such that whenever such error arises the app doesn't crashes instead it returns a null object. Which in th code indicates that the user hasn't made any contributions hence display the appropriate UI to the user.

Hope I am clear with whatever i said.
Still eager for valuable suggestions :)

File: app/build.gradle (hunk 1 above)
Comment (vanshikaarora) [reply app/build.gradle]: Fixed in the next commit

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java (hunk 2 above)
Comment (vanshikaarora) [reply app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: Fixed in the next commit

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java (hunk 4 above)
Comment (vanshikaarora) [reply app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: Fixed in the next commit

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java (hunk 5 above)
Comment (vanshikaarora) [reply app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: Fixed int the next commit

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java (hunk 6 above)
Comment (vanshikaarora) [reply app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: Fixed int the next commit

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java (hunk 7 above)
Comment (vanshikaarora) [reply app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: Fixed int the next commit

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java (hunk 8 above)
Comment (vanshikaarora) [reply app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: Fixed int the last commit

**Comment by maskaravivek** [comment]
@vanshikaarora Can you resolve the merge conflicts. :)

**Comment by vanshikaarora** [comment]
Yes sure I'll do that by tonight

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java (hunk 2 above)
Comment (neslihanturan) [reply app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: Please clear //onError(); line too

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java [hunk 11]
```diff
@@ -13,6 +14,7 @@
 import android.support.v7.app.AlertDialog;
 import android.support.v7.widget.Toolbar;
 import android.util.DisplayMetrics;
+import android.util.Log;
 import android.view.ContextThemeWrapper;
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: Please remove unused Log import

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java (hunk 6 above)
Comment (neslihanturan) [reply app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: Also this line: //levelNumber.setText("You haven't qualified any level yet");

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java [hunk 12]
```diff
@@ -388,10 +437,9 @@ private void launchAlert(String title, String message){
     private boolean checkAccount(){
         Account currentAccount = sessionManager.getCurrentAccount();
         if (currentAccount == null) {
-        Timber.d("Current account is null");
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: Why did you modify this line and remove Timber logs, it is valid.

Comment (neslihanturan) [reply app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: I see no need to modify this part at all

File: app/src/main/java/fr/free/nrw/commons/mwapi/ApacheHttpClientMediaWikiApi.java (hunk 9 above)
Comment (neslihanturan) [reply app/src/main/java/fr/free/nrw/commons/mwapi/ApacheHttpClientMediaWikiApi.java]: What about returning null and handling this null variable where it will be used?

File: gradle/wrapper/gradle-wrapper.properties [hunk 13]
```diff
@@ -1,6 +1,6 @@
-#Sat Nov 03 11:02:51 CET 2018
+##Sat Nov 03 11:02:51 CET 2018
```
Comment (neslihanturan) [inline gradle/wrapper/gradle-wrapper.properties]: This file shouldn't be changed.

File: app/src/main/res/values/strings.xml [hunk 14]
```diff
@@ -433,4 +433,7 @@ Upload your first media by touching the camera or gallery icon above.</string>
   <string name="display_location_permission_title">Display location permission</string>
   <string name="display_location_permission_explanation">Ask for location permission when needed for nearby notification card view feature.</string>
 
+  <string name="no_image">No images used</string>
+  <string name="no_image_reverted">No image reverted</string>
+  <string name="no_image_uploaded">No images Uploaded</string>
 </resources>
```
Comment (neslihanturan) [inline app/src/main/res/values/strings.xml]: I am not an English pro but "image" should be "images" and "Uploaded" -> "uploaded"

**Comment by neslihanturan** [comment]
Thanks  @vanshikaarora , we are almost there. I have added some more reviews, please consider them and fix conflicts.

File: gradle/wrapper/gradle-wrapper.properties (hunk 13 above)
Comment (vanshikaarora) [reply gradle/wrapper/gradle-wrapper.properties]: Hey @neslihanturan in response to your suggestion of returning a null instead of an empty object

I had already tried this before and on returning null the observable where this value is returned generates another exception since it can't handle a null response

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java (hunk 12 above)
Comment (vanshikaarora) [reply app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: I thought you have mentioned to remove all logs and comment statements but if these things are valid then i'll put them back

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java (hunk 11 above)
Comment (vanshikaarora) [reply app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: Done

File: app/src/main/res/values/strings.xml (hunk 14 above)
Comment (vanshikaarora) [reply app/src/main/res/values/strings.xml]: Yes you are right here. Things must be at perfection level. I'll do that

**Comment by neslihanturan** [comment]
> Hey @neslihanturan in response to your suggestion of returning a null instead of an empty object
> 
> I had already tried this before and on returning null the observable where this value is returned generates another exception since it can't handle a null response

I never tried returning null with observables but, can't we handle this exception too? For other developers it is easier to understand the reason of a null value.

File: gradle/wrapper/gradle-wrapper.properties [hunk 15]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.4-all.zip
```
Comment (neslihanturan) [inline gradle/wrapper/gradle-wrapper.properties]: This pages displays differences in the files between our repo and your PR branch. Since we want to keep this file unchanged, I would copy file content from repository and paste, so that there won't be any difference, and file will not appar here :)

**Comment by vanshikaarora** [comment]
Yes I even tried using try catch but that didn't worked out pretty well. If you have any better idea for that then i would definitely love to work

**Comment by neslihanturan** [comment]
You can fix other issues meanwhile:)

**Comment by vanshikaarora** [comment]
@neslihanturan Have you given a thought to it?

**Comment by maskaravivek** [comment]
@vanshikaarora Congrats for your first PR. :)

**Comment by vanshikaarora** [comment]
Thanks a lot :)
